apps/client/src/components/upload/LoadingProgress.vue

```diff
@@ -0,0 +1,172 @@
+<script lang="ts">
+import type { PropType } from 'vue';
+export interface ProgressRecord {
+    label: string;
+    progress: number;
```

Comment:
Changed to number enumeration.

Reply:
Fixed here: b9d7368b085da5085d4dad3be88e967dd43c502f

Reply:
For this I meant an actual typescript enum (https://www.typescriptlang.org/docs/handbook/enums.html).
But maybe an easier solution is the string union type:

e.g. 

`progress: 'failed' | 'not_started' | 'dispatched' | 'running' | 'finished'`

Right now this code is a bit hard to read (and maintain) with all the references to these special numbers. If we make this change it will be a lot easier to understand without needing to look back at your documentation.

This should also probably be saved as a type once and reused in uploadStore.ts.

Reply:
Fixed here: [5901b4a](https://github.com/visdesignlab/loonar/pull/40/commits/5901b4a97abe3eea2cd63ae5657bdd25cf6de797)

Reply:
Great! Could you also make the update for the uploadStore? I believe that had the same 5 numbers that matched?

Reply:
Yeah, I missed that that was also required haha. Fixed here: [6c1e2e9](https://github.com/visdesignlab/loonar/pull/40/commits/6c1e2e998ade872f026e5f1209acc36527f9afe6)

I made a progress type as well thats exported from the upload store, so it should be a bit more straightforward.